numpy/ma/tests/test_core.py

```diff
@@ -3138,14 +3133,14 @@ def test_inplace_floor_division_array_type(self):
 
     def test_inplace_division_scalar_type(self):
         # Test of inplace division
-        for t in self.othertypes:
-            with suppress_warnings() as sup:
-                sup.record(UserWarning)
```

Comment:
Could use `pytest.warns(UserWarning) as rec:` here to keep it closer to the old one?  These context inherit from `catch_warnings`, so you can add the `simplefilter()` inside of that without another indentation level.

Reply:
Ah `with pytest.WarningsRecorder() as rec` probably works, but yeah, nvm either way.  (it isn't a common pattern anyway and at that point I am not sure it's an improvement at all.).